plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ContractDescriptionConversionVisitor.kt

```diff
@@ -130,30 +145,36 @@ class ContractDescriptionConversionVisitor(
     ): ExpEmbedding {
         val param = callsEffect.valueParameterReference.accept(this, data)
         val callsFieldAccess = FieldAccess(param, SpecialFields.FunctionObjectCallCounterField)
+        val sourceRole = SourceRole.CallsInPlaceEffect(data.targetParameterFunction!!, callsEffect.kind)
```

Comment:
We already have access to the parameter here.  Why not use the same principle to obtain this information?  It seems simpler than passing it along through a different channel.

Reply:
I do agree, but I should find a way to access the function's symbol to get the parameter's one. One solution would be to pass the `FirFunctionSymbol` to `ContractDescriptionConversionVisitor` when instantiated, therefore removing it from the signatures of `getPreconditions/Postconditions`. WDYT?